### PR TITLE
Email Popout Compose View Uses Macro in Subject Line

### PR DESCRIPTION
**Pull request by cameronblaikie**
## Description

When using compose email from activities subpanel on cases the case name and number will be added to the Subject Line automatically and uses the macro so the CRM will parse it back to the case. This also means you can now send attachments using this method on Cases.

## How To Test This

1. Go to a case, then activities subpanel, then compose email.
2. See that the macro works as it should look like \[CASE:***NUMBER***\] ***CASE NAME*** 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by clemente-raposo** [comment]
@cameronblaikie @jack7anderson7 the commit message needs the issue id

**Comment by clemente-raposo** [comment]
@cameronblaikie @jack7anderson7 this will replace the subject for any email related with a case. 
I'm not sure we want to do that.

@mattlorimer something do discuss

**Comment by simonr44** [comment]
The intention here was to ensure emails sent manually from a case have the appropriate macro added to the subject automatically, in the same way the case updates do.

Replies to that email will then be picked up correctly by the inbound parser and attributed to the correct case.

There may be a better way to achieve this, but it should be quite flexible (i.e can be removed by the user) and ensures case email behaviour is consistent with case updates

File: modules/Emails/Email.php [hunk 1]
```diff
@@ -3489,7 +3489,13 @@ public function fill_in_additional_detail_fields()
             $this->status_name = $app_list_strings['dom_email_status'][$this->status];
         }
 
-        if (empty($this->name) && empty($_REQUEST['record']) && !empty($mod_strings['LBL_NO_SUBJECT'])) {
+        if($_REQUEST['targetModule'] === "Cases" && !empty($_REQUEST["ids"])) {
+            $caseId = $_REQUEST['ids'];
+            $aCase = BeanFactory::getBean('Cases', $caseId);
+
+            $this->name = str_replace('%1', $aCase->case_number, $aCase->getEmailSubjectMacro() . " " . $aCase->name);
```
Comment (clemente-raposo) [inline modules/Emails/Email.php:3496]: Hi @cameronblaikie,

Could we add some checks here please? I was thinking on the following ones:
- Check that `$_REQUEST['ids']` is a string and not an array
- Check if `$aCase = BeanFactory::getBean('Cases', $caseId);` returned a bean or not.

**Comment by serhiisamko091184** [comment]
Hello all! 
@cameronblaikie 
would you be so kind to have a look at last code review comment? 
Thank you very much! 

Regards,
Serhii

**Comment by serhiisamko091184** [comment]
Hello, @cameronblaikie 

The PR has been marked as stale because there has been no recent activity. It will be closed if no further activity occurs.
Thanks for your contributions.

Regards,
Serhii

**Comment by serhiisamko091184** [comment]
Hey @cameronblaikie 

Closing this PR due to inactivity. 
Thank you.

Regards,
Serhii

**Pull request closed without merging**